common/src/main/java/com/wynntils/screens/itemfilter/type/FilterWidgetFactory.java

```diff
@@ -0,0 +1,241 @@
+/*
+ * Copyright © Wynntils 2024.
+ * This file is released under LGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.screens.itemfilter.type;
+
+import com.wynntils.screens.itemfilter.ItemFilterScreen;
+import com.wynntils.screens.itemfilter.widgets.GeneralFilterWidget;
+import com.wynntils.screens.itemfilter.widgets.ProviderFilterListWidget;
+import com.wynntils.screens.itemfilter.widgets.StringFilterWidget;
+import com.wynntils.screens.itemfilter.widgets.numeric.InequalityCappedValueFilterWidget;
+import com.wynntils.screens.itemfilter.widgets.numeric.InequalityIntegerFilterWidget;
+import com.wynntils.screens.itemfilter.widgets.numeric.InequalityStatValueFilterWidget;
+import com.wynntils.screens.itemfilter.widgets.numeric.RangedCappedValueFilterWidget;
+import com.wynntils.screens.itemfilter.widgets.numeric.RangedIntegerFilterWidget;
+import com.wynntils.screens.itemfilter.widgets.numeric.RangedStatValueFilterWidget;
+import com.wynntils.screens.itemfilter.widgets.numeric.SingleCappedValueFilterWidget;
+import com.wynntils.screens.itemfilter.widgets.numeric.SingleIntegerFilterWidget;
+import com.wynntils.screens.itemfilter.widgets.numeric.SingleStatValueFilterWidget;
+import com.wynntils.services.itemfilter.filters.PercentageStatFilter;
+import com.wynntils.services.itemfilter.filters.RangedStatFilters;
+import com.wynntils.services.itemfilter.type.StatProviderAndFilterPair;
+import com.wynntils.services.itemfilter.type.StatValue;
+import com.wynntils.utils.type.CappedValue;
+import java.util.Map;
+import java.util.function.Function;
+
+public final class FilterWidgetFactory {
```

Comment:
Maybe rename to NumericFilterWIdgetFactory? I was looking for boolean/string types here. Or what if you moved them here?

Reply:
String is in the `createFilterWidget` method, though that's only for the providers without a list of valid inputs.

Boolean didn't seem worth adding to it as it's literally just

```
if (provider.getType().equals(Boolean.class)) {
    if (filterPairs.isEmpty()) {
        widgets.add(new BooleanFilterWidget(null, this));
    } else {
        widgets.add(new BooleanFilterWidget(filterPairs.get(0), this));
    }
} 
```

Reply:
What do you think about the rename, then?

Reply:
Probably could do with a more specific name, not quite sure right now I'll give it some thought

Reply:
I think it's best to just call it Numeric and take the string case out of it